ipld/get_shares.go

```diff
@@ -0,0 +1,101 @@
+package ipld
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gammazero/workerpool"
+	"github.com/ipfs/go-cid"
+	format "github.com/ipfs/go-ipld-format"
+)
+
+// NumWorkersLimit sets global limit for workers spawned by GetShares.
+// GetShares could be called MaxSquareSize(128) times per data square each spawning up to 128/2 goroutines and
+// altogether this is 8192. Considering there can be N blocks fetched at the same time, e.g. during catching up data
+// from the past, we multiply this number by the amount of allowed concurrent data square fetches(NumConcurrentSquares).
+// NOTE: This value only limits amount of simultaneously running workers that are spawned as the load increases and
+// are killed, once the load declines. Thanks to simple yet powerful workerpool library by @gammazero.
+// TODO(@Wondertan): This assumes we have parallelized DASer implemented. Sync the values once it is shipped.
+// TODO(@Wondertan): Allow configuration of values.
+var NumWorkersLimit = MaxSquareSize * MaxSquareSize / 2 * NumConcurrentSquares
+
+// NumConcurrentSquares limits the amount of squares that are fetched concurrently/simultaneously.
+var NumConcurrentSquares = 8
```

Comment:
I think the same feedback applies here: https://github.com/celestiaorg/celestia-node/pull/682/files#r867424676

Reply:
The same TODO I made applies here as well